test cache page allocation failure

rrrrrrmb opened the pull request:
This PR solves #442. This test creates an `entry` structure to initialize the cache and then creates more structures to  fill the cache until a new page allocation fails.

rrrrrrmb committed:
test cache page allocation failure

rrrrrrmb committed:
remove unused entry pointer

goatshriek reviewed (changes requested):
Your test works perfectly, and has in fact uncovered a memory leak!

The result of `realloc_mem` on line 70 of `src/cache.c` isn't dealt with properly in the case of failure. The easiest solution here would be to move the assignment currently on line 83 to before the alloc call, so that the cache has the new structure even in the case of an error. If you'd like to give that a shot please do, or if you see a better solution you can try that instead.

rrrrrrmb committed:
fix memory leak

this updates cache structure before the alloc_mem call so that the
structure is intact even in case of an allocation failure

rrrrrrmb commented:
Thanks for the hint ! I've made the suggested change and looks like that fixed it :)
